### ISSUE TITLE
Run-time dependency sdl2 found: NO (tried pkgconfig, config-tool and cmake)

### ISSUE THREAD

**Issue by urasssopretty** [mesonbuild/meson#12220]
**Describe the bug**
installing sdl2 from wrap works for linux but not for windows

**Output**
to cmd: meson setup --native-file .\src\native-windows.build .\build\debug\ .\src\

The Meson build system
Build type: native build
Project name: something
Project version: 0.0.0.1
C compiler for the host machine: C:\msys64\mingw64\bin\gcc.exe (gcc 13.1.0 "gcc.exe (Rev7, Built by MSYS2 project) 13.1.0")
C linker for the host machine: C:\msys64\mingw64\bin\gcc.exe ld.bfd 2.40
C++ compiler for the host machine: C:\msys64\mingw64\bin\g++.exe (gcc 13.1.0 "g++.exe (Rev7, Bui by MSYS2 project) 13.1.0")lt by MSYS2 project) 13.1.0")
C++ linker for the host machine: C:\msys64\mingw64\bin\g++.exe ld.bfd 2.40                      lt by MSYS2 project) 13.1.0")
Host machine cpu family: x86_64
Host machine cpu: x86_64
Found pkg-config: C:\msys64\mingw64\bin\pkg-config.EXE (1.9.5)
**sdl2-config found: NO**
Found CMake: D:\programms\cmake\bin\cmake.EXE (3.27.4)
**Run-time dependency sdl2 found: NO (tried pkgconfig, config-tool and cmake)**

src\meson.build:14:11: ERROR: Dependency "sdl2" not found, tried pkgconfig, config-tool and cmake     

**To Reproduce**
project(
  'something',
  'cpp', 'c',
  meson_version: '>= 1.2.0',
  version: '0.0.0.1',
  default_options: [
    'warning_level=1',
    'cpp_rtti=false', 
    'b_lto=true',
  ]
)

sdl2_dep = dependency('sdl2')
deps = [sdl2_dep]

include_dir = include_directories('includes')

sources = [
  'main.cc',
]

executable(
  'output',
  sources,
  build_by_default: true,
  win_subsystem: 'windows',
  dependencies: deps,
  include_directories: include_dir,
)

**native files for windows and linux**
[binaries]
c = ['ccache', '/usr/bin/clang']
cpp = ['ccache', '/usr/bin/clang++']

[built-in options]
buildtype = 'debug'
backend = 'ninja'
cpp_std = 'c++20'
c_std = 'c17'

///////
[binaries]
c = ['ccache', 'C:\msys64\mingw64\bin\gcc.exe']
cpp = ['ccache', 'C:\msys64\mingw64\bin\g++.exe']

[built-in options]
backend = 'ninja'
buildtype = 'debug'
cpp_std = 'c++20'
c_std = 'c17'

**Expected behavior**
I expect this file to work for both platforms

**system parameters**
it's native build for one computer
endeavouros / windows
Python 3.10.6
meson 1.2.1
ninja 1.11.1

**Comment by tristan957** [comment]
> sdl2-config found: NO

What does `which sdl2-config` point to on your Linux system?

**Comment by urasssopretty** [comment]
> > sdl2-config found: NO
> 
> What does `which sdl2-config` point to on your Linux system?

/usr/bin/sdl2-confif

**Comment by tristan957** [comment]
Is sdl2-config on your PATH on Windows?

**Comment by urasssopretty** [comment]
yes the path environment variable contains sdl2 config, I tried to indicate the path to the file itself and to the folder with it
![Untitled](https://github.com/mesonbuild/meson/assets/43519306/27506716-1ee3-4678-a0fe-e92e8a19426d)

**Comment by tristan957** [comment]
On Linux, does it get configured with cmake or config-tool? I have a feeling that there is some setting not being set for cmake to tell it where to find these cmake files. It sounds like you might need to set CMAKE_MODULE_PATH. https://mesonbuild.com/howtox.html#use-existing-findnamecmake-files

**Comment by tristan957** [comment]
`cmake_module_path: 'cmake'`. This needs to be pointing at the directory containing the cmake files in your picture

**Comment by urasssopretty** [comment]
oh im sry, like this?
![image](https://github.com/mesonbuild/meson/assets/43519306/0c6b4c45-d39a-4496-9cb5-0d110cfa6783)

**Comment by tristan957** [comment]
Maybe try `VulkanSDK\...\cmake`. Otherwise I think you are going to have to work on debugging this yourself. I have a feeling that this issue is probably specific to your system

**Comment by urasssopretty** [comment]
this doesn't work, thanks for trying to help

**Comment by tristan957** [comment]
No problem. Hope you figure it out :). Last chance. Does the dependency need to be called `SDL2` instead of `sdl2`? Can't remember CMake semantics

**Comment by urasssopretty** [comment]
ahahahhaah I tried this every time

**Comment by HenningScheufler** [comment]
```
cgns_dep = dependency('cgns',
                       method: 'cmake',
                       cmake_module_path : '/media/henning/ssd1linux/mambaforge/envs/cgns/lib/cmake/cgns')
```

```bash
ls /media/henning/ssd1linux/mambaforge/envs/cgns/lib/cmake/cgns
```
![grafik](https://github.com/mesonbuild/meson/assets/16381681/dd74925f-a6c4-461f-97df-4f402f9cc4c6)

I'm also unable to find the dependency and new to the meson and cmake build system. But there is a change that the cgns is not properly configured. I attached the content of the files
[cgns-targets.zip](https://github.com/mesonbuild/meson/files/12641188/cgns-targets.zip)


Could you help me?

**Comment by urasssopretty** [comment]
> ```
> cgns_dep = dependency('cgns',
>                        method: 'cmake',
>                        cmake_module_path : '/media/henning/ssd1linux/mambaforge/envs/cgns/lib/cmake/cgns')
> ```
> 
> ```shell
> ls /media/henning/ssd1linux/mambaforge/envs/cgns/lib/cmake/cgns
> ```
> 
> ![grafik](https://user-images.githubusercontent.com/16381681/268459771-dd74925f-a6c4-461f-97df-4f402f9cc4c6.png)
> 
> I'm also unable to find the dependency and new to the meson and cmake build system. But there is a chance that the cgns is not properly configured. I attached the content of the files [cgns-targets.zip](https://github.com/mesonbuild/meson/files/12641188/cgns-targets.zip)
> 
> Could you help me?

try someing like this
import cmake and add ur dependency like subproject
[cgns](https://mesonbuild.com/CMake-module.html#cmake-subprojects)

**Comment by HenningScheufler** [comment]
`cgns_dep2 = meson.get_compiler('cpp').find_library('cgns', required : true)`

i was able to find the library with this command but I'm uncertain what information the dependecy hold is there anyway to extract this information?
I tried:

`message('summary_value', cgns_dep2.get_compile_args())`

**Comment by tristan957** [comment]
the dep is essentially just the linker args. You can't know anything using find_library(). Only through dependency().
